wit/wasi-filesystem.wit.md

```diff
@@ -226,89 +226,87 @@ record dir-entry {
 }
 ```
 
-## `errno`
+## `error-code`
 ```wit
 /// Error codes returned by functions.
 /// Not all of these error codes are returned by the functions provided by this
 /// API; some are used in higher-level library layers, and others are provided
 /// merely for alignment with POSIX.
-enum errno {
+enum error-code {
     /// Permission denied.
     access,
     /// Resource unavailable, or operation would block.
     again,
     /// Connection already in progress.
     already,
     /// Bad descriptor.
-    badf,
+    bad-descriptor,
     /// Device or resource busy.
     busy,
     /// Resource deadlock would occur.
-    deadlk,
+    deadlock,
     /// Storage quota exceeded.
-    dquot,
+    quota,
     /// File exists.
     exist,
     /// File too large.
-    fbig,
+    file-too-large,
     /// Illegal byte sequence.
-    ilseq,
+    Illegal-byte-sequence,
     /// Operation in progress.
-    inprogress,
+    in-progress,
     /// Interrupted function.
-    intr,
+    interrupted,
     /// Invalid argument.
-    inval,
+    invalid,
     /// I/O error.
     io,
     /// Is a directory.
-    isdir,
+    is-directory,
     /// Too many levels of symbolic links.
     loop,
     /// Too many links.
-    mlink,
+    too-many-links,
     /// Message too large.
-    msgsize,
+    message-size,
     /// Filename too long.
-    nametoolong,
+    name-too-long,
     /// No such device.
-    nodev,
+    no-device,
     /// No such file or directory.
-    noent,
+    no-entry,
     /// No locks available.
-    nolck,
+    no-lock,
     /// Not enough space.
-    nomem,
+    insufficient-memory,
     /// No space left on device.
-    nospc,
-    /// Function not supported.
-    nosys,
```

Comment:
In this PR, I remove `nosys` (Function not supported) in favor of the new `unsupported` (Not supported). Another option if we want to not change the granualrity of error codes is to keep `nosys` and rename it to `function-unsupported`, but having both `unsupported` and `function-unsupported` looked a bit confusing to me, hence my proposal to translate both `nosys` and `notsup` to `unsupported` (but no strong take on that since I am not a POSIX expert). I also removed the comment about socket since I think it is now out of the scope.